### PR TITLE
Add getting started docs

### LINKED ISSUE #33: Add documentation for getting started
The existing project documentation doesn't cover getting started with the cookiecutter, and assumes the presence of certain required software (`cookiecutter`, for instance). In a similar vein to how the [UK provides a getting started guide](https://github.com/best-practice-and-impact/govcookiecutter), there needs to be one for users of this as well.

Since we're using `conda` as the base package it isn't clear if required software should be in a requirements file for pip or a conda environment specification.

Known required packages:

* cookiecutter
* pre-commit

### PR DESCRIPTION
**Pull request by goatsweater**
Brings in the usage documentation developed as part of the Hydra project and other experiences.

Closes #33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +18,12 @@ Unless otherwise noted, the source code of this project is covered under Crown C
 
 The Canada wordmark and related graphics associated with this distribution are protected under trademark law and copyright law. No permission is granted to use them outside the parameters of the Government of Canada's corporate identity program. For more information, see [Federal identity requirements](https://www.canada.ca/en/treasury-board-secretariat/topics/government-communications/federal-identity-requirements.html).
 
+### Acknowledgements
+
+This templates is inspired by the work of the [DrivenData cookiecutter data science project](http://drivendata.github.io/cookiecutter-data-science/),
```
Comment (goatsweater) [inline README.md:23]:
```suggestion
This template is inspired by the work of the [DrivenData cookiecutter data science project](http://drivendata.github.io/cookiecutter-data-science/),
```

File: docs/usage.rst [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+Initial setup
+-------------
+
+To set yourself up for success, ensure your system is ready to use the template.
+
+Required software
+^^^^^^^^^^^^^^^^^
+
+This project is based on `cookiecutter`_, but we recommend `cruft`_ to help keep your
+project in sync as the templates evolve to provide additional usaful features.
```
Comment (goatsweater) [inline docs/usage.rst:17]:
```suggestion
project in sync as the templates evolve to provide additional useful features.
```

File: docs/usage.rst [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    $ conda activate <repo name>
+
+Several git hooks have been pre-configured for you. ``pre-commit`` is used to make configuring
+them easy:
+
+.. code-block:: sh
+
+    $ pre-commit install
+
+Now the hooks will run autatically every time you run ``git commit``.
```
Comment (goatsweater) [inline docs/usage.rst:91]:
```suggestion
Now the hooks will run automatically every time you run ``git commit``.
```

**Review by ToucheSir — APPROVED** [review]
Nice to have some usage notes based on real-world experience!

**Pull request merged**